source/common/network/socket_option_impl.cc

```diff
@@ -23,6 +23,18 @@ bool SocketOptionImpl::setOption(Socket& socket,
   return true;
 }
 
+absl::optional<Socket::Option::Details>
+SocketOptionImpl::getOptionDetails(const Socket&,
+                                   envoy::api::v2::core::SocketOption::SocketState state) const {
+  if (state != in_state_) {
+    return absl::nullopt;
+  }
+  Socket::Option::Details info;
+  info.name_ = optname_;
+  info.value_ = value_;
+  return absl::optional<Option::Details>(std::move(info));
+}
```

Comment:
If I read the code correctly, `optname_` might be an absl::nullopt if the OS doesn't support the option. (E.g. `ENVOY_SOCKET_IP_TRANSPARENT` might be `SocketOptionName()` on some systems.

Should we return absl::nullopt here in that case?

Reply:
Hmm... That's a good question. Is an option not being supported equivalent to it not existing at all? Likely. It'd mean that any UT using this would need to protect against the option not being supported, but that's probably fine, since anything that is searching a list of options that are all unsupported would likely mess up anyway. I'll make this change.